evm/db/journal.py

```diff
@@ -1,226 +1,260 @@
-from typing import Dict  # noqa: F401
+import collections
+from typing import Dict, Union  # noqa: F401
 import uuid
 
 from cytoolz import (
+    first,
     merge,
+    last,
 )
 
 from evm.db.backends.base import BaseDB
 from evm.exceptions import ValidationError
 
 
-class Journal(object):
+class DeletedEntry:
+    pass
+
+
+DELETED_ENTRY = DeletedEntry()
```

Comment:
Purely a personal style thing, but you may like `DELETED_ENTRY = object()` in order to skip the class def'n.

Reply:
I like it. Give me that lightweight JavaScript feeling. I'm on mobile. Will make that change later and then squash and merge.

Reply:
Actually, I think I'm gonna keep the `DeletedEntry` because that's a bit more strict. Can't just push an arbitrary `object` into a `Dict[bytes, Union[bytes, DeletedEntry]]`. One could achieve the same strictness with a `NewType` but at this point I think the extra class is just fine.

Reply:
Yup, my style on that was developed in a dynamic type context. Makes total sense. :+1: